src/pages/Account/Tabs/ModulesTab/Contract.tsx

```diff
@@ -228,15 +228,20 @@ function ReadContractForm({
 }) {
   const [state] = useGlobalState();
   const [result, setResult] = useState<Types.MoveValue[]>([]);
+  const [errMsg, setErrMsg] = useState<string>();
 
   const onSubmit: SubmitHandler<ContractFormType> = async (data) => {
     const viewRequest: Types.ViewRequest = {
-      function: `${module.address}::${module.name}::${fn.name}`,
+      function: `${module.address}::${module.name}::${fn.name}!`,
       type_arguments: data.typeArgs,
       arguments: data.args,
     };
-    const result = await view(viewRequest, state.network_value);
-    setResult(result);
+    try {
+      const result = await view(viewRequest, state.network_value);
+      setResult(result);
+    } catch (e: any) {
+      setErrMsg(e.message ?? String(e));
+    }
```

Comment:
this error msg persist even if i switch function names

Reply:
@jillxuu can you help me understand why? after switching the function name, i thought the ReadContractForm will re-render since props.fn change, therefore reset all internal states. however it doesn't

Reply:
it seems like the entire component's not re-rendering when `fn` changed, not sure exactly why. but a potential fix is to add 
```
 useEffect(() => {
    setErrMsg("");
  }, [fn]);
```
to force re-render when `fn` changes.